components/menu/menu.tsx

```diff
@@ -0,0 +1,259 @@
+                        menuId: props.menu.id,
+                        menuAriaLabel: props.menu?.['aria-label'] ?? '',
+                        children: props.children,
+                    },
+                }),
+            );
+        } else {
+            setAnchorElement(event.currentTarget);
+        }
+    }
+
+    function handleMenuButtonMouseDown() {
+        // This is needed to prevent focus-visible being set on clicking menuitems with mouse
+        setDisableAutoFocusItem(true);
+    }
+
+    function renderMenuButton() {
+        // We construct the menu button so we can set onClick correctly here to support both web and mobile view
+        const triggerElement = (
+            <button
+                id={props.menuButton.id}
+                aria-controls={props.menu.id}
+                aria-haspopup={true}
+                aria-expanded={isMenuOpen}
+                aria-label={props.menuButton?.['aria-label'] ?? ''}
+                className={props.menuButton?.class ?? ''}
+                onClick={handleMenuButtonClick}
+                onMouseDown={handleMenuButtonMouseDown}
+            >
+                {props.menuButton.children}
+            </button>
+        );
+
+        if (props.menuButtonTooltip && props.menuButtonTooltip.text && !isMobileView) {
+            return (
+                <OverlayTrigger
+                    delayShow={OVERLAY_TIME_DELAY}
+                    placement={props?.menuButtonTooltip?.placement ?? 'top'}
+                    overlay={
+                        <Tooltip
+                            id={props.menuButtonTooltip.id}
+                            className={props.menuButtonTooltip?.class ?? ''}
+                        >
+                            {props.menuButtonTooltip.text}
+                        </Tooltip>
+                    }
+                    disabled={isMenuOpen}
+                >
+                    {triggerElement}
+                </OverlayTrigger>
+            );
+        }
+
+        return triggerElement;
+    }
+
+    useEffect(() => {
+        if (props.menu.onToggle) {
+            props.menu.onToggle(isMenuOpen);
+        }
+    }, [isMenuOpen]);
+
+    if (isMobileView) {
+        // In mobile view, the menu is rendered as a modal
+        return renderMenuButton();
+    }
+
+    return (
+        <CompassDesignProvider theme={theme}>
+            {renderMenuButton()}
+            <MuiMenuStyled
+                anchorEl={anchorElement}
+                open={isMenuOpen}
+                onClose={handleMenuClose}
+                onClick={handleMenuClick}
+                onKeyDown={handleMenuKeyDown}
+                className={A11yClassNames.POPUP}
+                disableAutoFocusItem={disableAutoFocusItem} // This is not anti-pattern, see handleMenuButtonMouseDown
+                MenuListProps={{
+                    id: props.menu.id,
+                    'aria-label': props.menu?.['aria-label'] ?? '',
```

Comment:
`??` not needed here

Reply:
i am not how the MenuListProps handles aria-label undefined. for normal components i would have just done that, but for this i had to do this